subprojects/packagefiles/glfw/meson.build

```diff
@@ -0,0 +1,259 @@
+    deps += dependency(w, required : true)
+  endforeach
+  wl_protocols = dependency('wayland-protocols')
+  wl_dir = wl_protocols.get_variable(pkgconfig : 'pkgdatadir')
+  wl_scan_dep = dependency('wayland-scanner'
+                          , required : false
+                          , native : true)
+  if wl_scan_dep.found()
+    wl_scanner = find_program(wl_scan_dep.get_variable(pkgconfig : 'wayland_scanner'), native : true)
+  else
+    wl_scanner = find_program('wayland-scanner', native : true)
+  endif
+  s_wayland = []
+  protocols = [['idle-inhibit', 'unstable']
+              ,['pointer-constraints', 'unstable']
+              ,['relative-pointer', 'unstable']
+              ,['viewporter', 'stable']
+              ,['xdg-decoration', 'unstable']
+              ,['xdg-shell', 'stable']]
+  foreach p : protocols
+    x_path = join_paths(wl_dir, p[1], p[0], p[0])
+    if p[1] == 'stable'
+      x_name = p[0]
+      xml = files(x_path + '.xml')
+    else
+      x_name = p[0] + '-unstable-v1'
+      xml = files(x_path + '-unstable-v1.xml')
+    endif
+    foreach f : ['client-header', 'server-header', 'private-code']
+      if (f == 'client-header' and p[0] == 'xdg-decoration')
+        outfile = 'wayland-' + p[0] + '-client-protocol.h'
+      elif f == 'client-header'
+        outfile = 'wayland-' + x_name + '-client-protocol.h'
+      elif f == 'server-header'
+        outfile = 'wayland-' + p[0] + '-server-protocol.h'
+      elif (f == 'private-code' and p[0] == 'xdg-decoration')
+        outfile = 'wayland-' + p[0] + '-protocol.c'
+      else
+        outfile = 'wayland-' + x_name + '-protocol.c'
+      endif
+      s_wayland += custom_target('@0@ @1@'.format(p[0], f)
+                               , command : ['wayland-scanner', f, '@INPUT@', '@OUTPUT@']
+                               , input : xml
+                               , output : outfile)
+    endforeach
+  endforeach
+#  Win32
+elif opt_display == 'win32'
+  #——TODO:
+  # ☐ Implement Win32 build
+  error('Win32 build is not currently implemented.')
+#  Cocoa
+elif opt_display == 'cocoa'
+  #——TODO:
+  # ☐ Implement Cocoa build
+  error('Cocoa build is not currently implemented.')
+endif
+#@ System Dependencies
+if (sys_os == 'linux' or sys_os == 'sunos')
+  foreach l : ['m', 'rt', 'dl']
+    deps += sys_cc.find_library(l, required : true)
+  endforeach
+  deps += dependency('threads')
+elif sys_os == bsd
+  foreach l : ['m', 'rt']
+    deps += sys_cc.find_library(l, required : true)
+  endforeach
+  deps += dependency('threads')
+endif
+
+
+### Configuration ###
+cfg_data = configuration_data()
+foreach d : ['osmesa', 'x11', 'wayland', 'win32', 'cocoa']
+  cfg_data.set('_GLFW_' + d.to_upper(), opt_display == d)
+endforeach
+cfg_file = configure_file(configuration : cfg_data
+                         ,output : 'glfw_config.h')
+
+
+### Targets ###
+#@ Install key
+I = opt_install
```

Comment:
Why is this defined statically once, and then reassigned to another variable? Just directly assign it. But really this is very inscrutable.

Reply:
Why, indeed? As part of your first comment about inlining variables used once, I went through and removed occurrences like this.